code/bml_builder.js

```diff
@@ -0,0 +1,87 @@
+(function() {
+  // Live version
+  let HOST = 'https://r.keymanweb.com';
+
+  // // Debug / dev version
+  // let HOST = 'http://r.keymanweb.com.local';
+
+  function construct_bookmarklet_stylesheet() {
```

Comment:
Why don't we just do this as a .css? We already do `<script src='bml_builder.js'></script>` so it makes sense to add `<link href='bml_builder.css'>` as well.

Reply:
I'm thinking ahead to an `esbuild`-based version of this that could bundle _everything_ into a single (.js) file.  After perusing the documentation, it doesn't seem like esbuild supports bundling .css files into .js files, though images can be bundled into either.  (It _does_ support bundling css as a final product, though!)

So, to facilitate a single point of linkage on pages if and when we decide to `esbuild` bundle, I felt it appropriate to embed the CSS within the JS now, to reduce the changeset needed later.

Reply:
AFAICT, esbuild does support bundling of css? https://esbuild.github.io/content-types/#css But I'm not sure what distinction you are making between bundling into a single (.js) file vs bundling css as a final product?

But even if it doesn't, I feel like embedding css into js for the sake of a potential future redesign is not ideal.

Given there are only two places we are using the bookmarklet builder, it doesn't seem like a whole lot of pain to fix them both up if needed in the future.

Reply:
Granted, I'm drawing inferences here, but note the very next section: https://esbuild.github.io/content-types/#css-from-js

> When you do this, esbuild will gather all CSS files referenced from a given entry point and bundle it into a sibling CSS output file next to the JavaScript output file for that JavaScript entry point.

To me, this implies that it will not bundle a CSS file _into_ a JS file.  It's perfectly happy to bundle CSS files together - like when one `@import`s others - and this is the use case talked about in the section you linked.  Also note that your linked section results in a `.css` output - not `.js`.

Reply:
> Given there are only two places we are using the bookmarklet builder, it doesn't seem like a whole lot of pain to fix them both up if needed in the future.

That's fair.  I was just thinking it'd be good practice for bundling in CSS stuff on a small scale before possibly doing that in something much larger - such as KMW itself.  Then again... there's so much CSS for KMW that we'd probably want a way to keep it as a separate .css file there, too.

Reply:
Yeah, I think one of the big selling points for bundling sub-resources into a .js was to avoid the performance cost from multiple connections to the server. Given recent HTTP versions avoid the downsides, people are now going away from extreme bundling once more, which is a good thing I think!

(But now I can't find any links though I read some articles recently!)